.github/workflows/package.yaml

```
@@ -0,0 +1,120 @@
+    - master
+    tags:
+    - '*'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  all:
+    name: All
+
+    strategy:
+      matrix:
+        target:
+        - aarch64-unknown-linux-gnu
+        - armv7-unknown-linux-gnueabihf
+        - x86_64-apple-darwin
+        - x86_64-pc-windows-msvc
+        - x86_64-unknown-linux-musl
+        include:
+        - target: aarch64-apple-darwin
+          os: macos-latest
+          target_rustflags: '--codegen target-feature=+crt-static'
+          cross_compiling: true
+        - target: aarch64-unknown-linux-gnu
+          os: ubuntu-latest
+          cross_compiling: true
+          target_rustflags: '--codegen linker=aarch64-linux-gnu-gcc'
+        - target: armv7-unknown-linux-gnueabihf
+          os: ubuntu-latest
+          cross_compiling: true
+          target_rustflags: '--codegen linker=arm-linux-gnueabihf-gcc'
+        - target: x86_64-apple-darwin
+          os: macos-latest
+          cross_compiling: false
+          target_rustflags: '--codegen target-feature=+crt-static'
+        - target: x86_64-pc-windows-msvc
+          os: windows-latest
+          cross_compiling: false
+          target_rustflags: '--codegen target-feature=+crt-static'
+        - target: x86_64-unknown-linux-musl
+          os: ubuntu-latest
+          cross_compiling: false
+          target_rustflags: '--codegen target-feature=+crt-static'
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Make openssl-src Use Strawberry Perl
+      if: matrix.os == 'windows-latest'
+      run: echo OPENSSL_SRC_PERL=C:/Strawberry/perl/bin/perl >> $GITHUB_ENV
+
+    - name: Install Rust Toolchain Components
+      uses: actions-rs/toolchain@v1
+      with:
+        override: true
+        target: ${{ matrix.target }}
+        toolchain: stable
+
+    - name: Update Apt
+      if: ${{ matrix.os == 'ubuntu-latest' }}
+      run: |
+        sudo apt-get update
+
+    - name: Install AArch64 Toolchain
+      if: ${{ matrix.target == 'aarch64-unknown-linux-gnu' }}
+      run: |
+        sudo apt-get install gcc-aarch64-linux-gnu
+
+    - name: Install ARM7 Toolchain
+      if: ${{ matrix.target == 'armv7-unknown-linux-gnueabihf' }}
+      run: |
+        sudo apt-get install gcc-arm-linux-gnueabihf
+
+    - name: Install Musl
+      if: ${{ matrix.target == 'x86_64-unknown-linux-musl' }}
+      run: |
+        sudo apt-get install musl-tools
+
+    - name: Test
+      if: '!matrix.cross_compiling'
+      run: |
+        cargo test --all --all-features
+        cargo test --test integration
+
+    - name: Package
+      id: package
+      env:
+        TARGET: ${{ matrix.target }}
+        REF: ${{ github.ref }}
+        OS: ${{ matrix.os }}
+        TARGET_RUSTFLAGS: ${{ matrix.target_rustflags }}
+      run: ./bin/package
+
+    - name: Release Check
+      id: release
+      run: |
+        if [[ '${{ github.ref }}' =~ '^refs/tags/[[:digit:]]+[.][[:digit:]]+[.][[:digit:]]+$' ]]; then
```

```suggestion
        if [[ '${{ github.ref }}' =~ ^refs/tags/[[:digit:]]+[.][[:digit:]]+[.][[:digit:]]+$ ]]; then
```
Do we need quotes here?

---

Good catch! This works on zsh but not on bash.